### ISSUE TITLE
bcpg-jdk18-on 1.72.1 imports wrong version of packages

### ISSUE THREAD

**Issue by grgrzybek** [bcgit/bc-java#1278]
In Maven Central the only `org.bouncycastle` library with 1.72.1 version is `bcpg-jdk18-on`, however it contains these OSGi headers:
```
Import-Package: \
 org.bouncycastle.asn1;version="1.72.1",\
 org.bouncycastle.asn1.edec;version="1.72.1",\
 org.bouncycastle.asn1.gnu;version="1.72.1",\
 org.bouncycastle.asn1.nist;version="1.72.1",\
 org.bouncycastle.asn1.pkcs;version="1.72.1",\
 org.bouncycastle.asn1.sec;version="1.72.1",\
 org.bouncycastle.asn1.teletrust;version="1.72.1",\
 org.bouncycastle.asn1.x509;version="1.72.1",\
 org.bouncycastle.asn1.x9;version="1.72.1",\
 org.bouncycastle.crypto;version="1.72.1",\
...
```

At first glance (it's not a Maven build...) I can't find where BC generates the OSGi headers, but the imports should use `1.72` version, not `1.72.1` for packages outside of `bcpg-jdk18-on`...

**Comment by dghgit** [comment]
Sigh... I actually thought I'd got that. Sorry. This jar doesn't need to be signed so it is easy for you to fix it by hand. If you need a release on maven, I'll need to do a 1.72.2 - it's not possible to change a jar after the fact.

**Comment by grgrzybek** [comment]
Thanks for response - actually, we can fix it at Karaf level (because BC is provided as system library with OSGi boot delegation), so it was a matter of exporting additional (to `1.72`) version of packages imported by `bcpg-jdk18-on` in version `1.72.1`.

While it's understandable (from `bnd` configuration) to have imports `[1.72.1,2)` (or simply `1.72.1` which is equivalent of `[1.72.1,∞)`), it requires release of all the artifacts at once.

I'm not a gradle expert (I have long time trust towards Maven builds, which are much more maintainable in the long term) so I don't know exactly what would need to be changed ;).

**Comment by dghgit** [comment]
So, something else has shown up.  Try https://www.bouncycastle.org/bcpg-jdk18on-1.72.2.zip and let me know how it goes. I think it will fix your issue as well.

**Comment by grgrzybek** [comment]
Looks good now. I see proper versions imported:
```
org.bouncycastle.asn1;version="1.72"
org.bouncycastle.asn1.edec;version="1.72"
org.bouncycastle.asn1.gnu;version="1.72"
org.bouncycastle.asn1.nist;version="1.72"
org.bouncycastle.asn1.pkcs;version="1.72"
org.bouncycastle.asn1.sec;version="1.72"
org.bouncycastle.asn1.teletrust;version="1.72"
org.bouncycastle.asn1.x509;version="1.72"
org.bouncycastle.asn1.x9;version="1.72"
org.bouncycastle.crypto;version="1.72"
org.bouncycastle.crypto.agreement;version="1.72"
org.bouncycastle.crypto.digests;version="1.72"
org.bouncycastle.crypto.ec;version="1.72"
org.bouncycastle.crypto.encodings;version="1.72"
org.bouncycastle.crypto.engines;version="1.72"
org.bouncycastle.crypto.generators;version="1.72"
org.bouncycastle.crypto.io;version="1.72"
org.bouncycastle.crypto.modes;version="1.72"
org.bouncycastle.crypto.params;version="1.72"
org.bouncycastle.crypto.signers;version="1.72"
org.bouncycastle.crypto.util;version="1.72"
org.bouncycastle.jcajce.io;version="1.72"
org.bouncycastle.jcajce.spec;version="1.72"
org.bouncycastle.jcajce.util;version="1.72"
org.bouncycastle.jce.provider;version="1.72"
org.bouncycastle.math.ec;version="1.72"
org.bouncycastle.math.ec.rfc8032;version="1.72"
org.bouncycastle.util;version="1.72"
org.bouncycastle.util.encoders;version="1.72"
org.bouncycastle.util.io;version="1.72"
```

and the exports are at version `1.72.2`.

BTW - where in the gradle build are the OSGi manifests generated? All I found was https://github.com/bcgit/bc-java/blob/master/add_module.sh#L105-L109 ...

**Comment by dghgit** [comment]
Thanks for the report. I've published a 1.72.2 to maven and added it to the latest releases page.
